lib/src/wolt_modal_sheet_route.dart

```diff
@@ -143,37 +107,29 @@ class WoltModalSheetRoute<T> extends PageRoute<T> {
     Animation<double> secondaryAnimation,
     Widget child,
   ) {
-    final modalType = _modalTypeBuilder(context);
-    const easeCurve = Curves.ease;
-    switch (modalType) {
-      case WoltModalType.bottomSheet:
-        return _bottomSheetTransitionAnimation ??
-            SlideTransition(
-              position: animation.drive(
-                Tween(
-                  begin: const Offset(0.0, 1.0),
-                  end: Offset.zero,
-                ).chain(CurveTween(curve: easeCurve)),
-              ),
-              child: child,
-            );
-      case WoltModalType.dialog:
-        return _dialogTransitionAnimation ??
-            ScaleTransition(
-              scale: animation.drive(Tween(
-                begin: 0.9,
-                end: 1.0,
-              ).chain(CurveTween(curve: easeCurve))),
-              child: child,
-            );
-    }
+    final modalType = _determineCurrentModalType(context);
+    return modalType.buildTransitions(
+        context, animation, secondaryAnimation, child);
   }
 
   @override
-  Color? get barrierColor => modalBarrierColor ?? Colors.black54;
+  Color? get barrierColor {
+    final context = navigator!.context;
+    final themeData = Theme.of(context).extension<WoltModalSheetThemeData>();
+    final defaultThemeData = WoltModalSheetDefaultThemeData(context);
+    return modalBarrierColor ??
+        themeData?.modalBarrierColor ??
+        defaultThemeData.modalBarrierColor;
+  }
 
   @override
-  Duration get transitionDuration => _transitionDuration;
+  Duration get transitionDuration {
+    return _determineCurrentModalType(navigator!.context).transitionDuration;
+  }
+
+  @override
+  Duration get reverseTransitionDuration =>
+      _determineCurrentModalType(navigator!.context).reverseTransitionDuration;
```

Comment:
Minor nit: `transitionDuration` uses curly braces and `reverseTransitionDuration` uses return operator. The package has been using return operator so we should keep it consistent.

Reply:
done ffc63266f27c38e080f613f4d6361fa3c2cf1010